ostd/src/sync/rwlock.rs

```diff
@@ -99,7 +92,10 @@ use crate::{
 ///     assert_eq!(*w2, 7);
 /// }   // write lock is dropped at this point
 /// ```
-pub struct RwLock<T: ?Sized> {
+///
+/// [`SpinLock`]: super::SpinLock
+pub struct RwLock<T: ?Sized, Guard: Guardian = PreemptDisabled> {
```

Comment:
The best practices of using trait bounds is to add them when they are used. If you read Rust std's code, you will notice that no trait bounds are given in generic struct definitions. This is because when we define a struct, we won't be able use any methods provided by any given trait bounds. We can only use them in the implementation blocks. So here remove the `Guardian` bound. Add the `G: Guardian` bound to implementation blocks only when the bound makes a difference.